src/libpmemfile-posix/file.c

```diff
@@ -338,6 +338,11 @@ _pmemfile_openat(PMEMfilepool *pfp, struct pmemfile_vinode *dir,
 			goto end;
 		}
 
+		if ((flags & PMEMFILE_O_RDWR) && vinode_is_dir(vinode)) {
```

Comment:
Permission part of "flags" is not really a bit mask. You have to first mask O_ACCMODE and then check for equality (see other uses of O_RDWR).

This code should also handle O_WRONLY.